soccer/gameplay/plays/testing/four_courners_pass.py

```diff
@@ -0,0 +1,201 @@
+		# four corners.
+		if (self.has_subbehavior_with_name('chasing')) :
+			self.chaser.pos = self.cut_of_pos(main.ball().pos)
+
+		#draw the four courner field
+		# takes in the square points and form lines and create a square on the field
+		for i in range(len(self.square_points)) :
+			main.system_state().draw_line(robocup.Line(self.square_points[i], 
+			self.square_points[(i + 1) % 4]), (135, 0, 255), "Square")
+
+
+	def on_enter_setup(self):
+		# find where we will consider the ball is closest too and passing from
+		closestPt = min(self.square_points, key=lambda pt: pt.dist_to(main.ball().pos))
+
+		closestPtIdx = self.square_points.index(closestPt)
+
+		#points that the other robots can move to
+		self.otherPts = list(self.square_points)
+		self.otherPts.remove(closestPt)
+
+		# remove the point that's diagonal of the closest point
+		farthestPt = max(self.otherPts,
+						key=lambda pt: pt.dist_to(main.ball().pos))
+
+		self.otherPts.remove(farthestPt)
+
+		# decide which direction you're passing to
+		self.direction = self.safer_pass()
+		# move the other two robots to the other point locations
+		self.add_subbehavior(skills.move.Move(self.square_points[(closestPtIdx + self.direction) % 4]), 'move1', required = False)
+		# send the closest robot to capture the ball
+		self.add_subbehavior(skills.capture.Capture(), 'capture')
+
+	def on_exit_setup(self):
+		self.remove_subbehavior('move1')
+		self.remove_subbehavior('capture')
+
+	def on_enter_passing(self):
+		# get the point passing from
+		kickFrom = min(self.square_points,
+					   key=lambda pt: pt.dist_to(main.ball().pos))
+		# get the index of where you're passing from
+		kickFromIdx = self.square_points.index(kickFrom)
+		# based on which direction you're passing to pick that index
+		kickToIdx = (kickFromIdx + self.direction) % len(self.square_points)
+		# get that point
+		kickToPt = self.square_points[kickToIdx]
+
+		# get ready to pass the ball
+		passing_action = tactics.coordinated_pass.CoordinatedPass(kickToPt)
+		# while pass is preparing get ready for the next pass
+		premove = skills.move.Move(self.square_points[(kickToIdx + self.direction) % len(self.square_points)])
+
+		self.add_subbehavior(passing_action, 'passer', required = True, priority = 10)
+		self.add_subbehavior(premove, 'premove', required = True, priority = 2)
+
+
+		# add a robot to chase after the ball.
+		if (not self.has_subbehavior_with_name('chasing')):
+			self.chaser = skills.move.Move(self.cut_of_pos(main.ball().pos))
+			self.add_subbehavior(self.chaser, 'chasing', required = True, priority = 2)
+
+		# if we have too many robots isolate one of the robots so they don't help in the play
+		if (not self.has_subbehavior_with_name('iso') and len(main.our_robots()) == 6) :
+			self.iso = skills.move.Move(robocup.Point(-constants.Field.Width/2, 0))
+			self.add_subbehavior(self.iso,'iso', required = True, priority = 1)
```

Comment:
A better way of doing this is to take the number of robots currently on our team and then creating a move subbehavior for each one not required. This could even happen in the constructor of the play

Reply:
You could probably even do the lineup tactic and just make the line somewhere out of the way

Reply:
That's probably a smarter idea how would I give them labels?

Reply:
Labels? Are you talking about the string part of the arg?

If so see here https://github.com/RoboJackets/robocup-software/blob/master/soccer/gameplay/tactics/line_up.py#L60